update pipelines benefits (#4031)
circleci/circleci-docs: jekyll/_cci2/api-intro.md

### Pipelines

> The CircleCI API v2 includes the use of pipelines to assist you in triggering workflows from the CircleCI API. By enabling pipelines, you can take advantage of the following:

Reviewer: ```suggestion
The CircleCI API v2 requires you to enable pipelines on your project. By enabling pipelines, you can also take advantage of the following:
```

Author: Good call!

> The CircleCI API v2 includes the use of pipelines to assist you in triggering workflows from the CircleCI API. By enabling pipelines, you can take advantage of specific API use cases such as:

Author: ```suggestion
The CircleCI API v2 includes the use of pipelines to assist you in triggering workflows from the CircleCI API. By enabling pipelines, you can take advantage of the following:
```
